src/toil/job.py

```diff
@@ -429,41 +429,45 @@ def asyncWrite():
 
         def getLocalTempDir(self):
             """
-            Get a new local temporary directory. This directory will exist for the 
+            Get the absolute path to a new local temporary directory. 
+            This directory will exist for the 
             duration of the job only, and is guaranteed to be deleted once
             the job terminates, removing all files it contains recursively. 
             """
-            return tempfile.mkdtemp(prefix="t", dir=self.localTempDir)
+            return os.path.abspath(tempfile.mkdtemp(prefix="t", dir=self.localTempDir))
```

Comment:
This shouldn't be necessary. We should just make sure that locaTempDir is absolute.